tools-local/Microsoft.ML.InternalCodeAnalyzer/BestFriendAnalyzer.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.ML.InternalCodeAnalyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class BestFriendAnalyzer : DiagnosticAnalyzer
+    {
+        private const string Category = "Access";
+        internal const string DiagnosticId = "MSML_NoBestFriendInternal";
+
+        private const string Title = "Cross-assembly internal access requires.";
+        private const string Format = "Access of '{0}' is a cross assembly internal " +
+            "reference, and the declaring assembly wants these accesses to be on something " +
+            "with the attribute " + AttributeName + ".";
+        private const string Description =
+            "The ML.NET .";
```

Comment:
Is this like `THE Ohio State University"? 😉

Reply:
*Exactly* like that, yes.